book/src/leader-rotation.md

```diff
@@ -3,17 +3,76 @@
 At any given moment, a cluster expects only one fullnode to produce ledger
 entries. By having only one leader at a time, all validators are able to replay
 identical copies of the ledger. The drawback of only one leader at a time,
-however, is that a malicious leader is cabable of censoring votes and
+however, is that a malicious leader is capable of censoring votes and
 transactions. Since censoring cannot be distinguished from the network dropping
 packets, the cluster cannot simply elect a single node to hold the leader role
-indefinitely. Instead, the cluster minimizes the influence of a malcioius
+indefinitely. Instead, the cluster minimizes the influence of a malicious
 leader by rotating which node takes the lead.
 
 Each validator selects the expected leader using the same algorithm, described
 below. When the validator receives a new signed ledger entry, it can be certain
 that entry was produced by the expected leader.
 
-## Leader Schedule Generation
+## Leader Schedule Rotation
+
+A block's slot is valid if it is generated by the leader computed from a Leader
+Schedule in the block that crossed the epoch slot boundary at least 1 epoch ago.
+
+Leader schedule is generated every epoch when a fork crosses the epoch boundary.
```

Comment:
a fork -> the slot height

Reply:
@garious, a block i think is more appropriate.  since many blocks can cross the same epoch boundary

Reply:
I had hoped we can leave blocks out of this. Blocks are per-fork. My understanding was the leader scheduler needs to be per-validator, not per-fork, so that we can filter out blocks from an unexpected leader.

Reply:
@garious But the event that causes a Leader Schedule to be generated is a proposed block that crosses the epoch boundary.  Any block can cross the epoch boundary, and therefore many Leader Schedulers are possible until a block is committed to root by the validator.

Reply:
What do you think about only generating leader schedules from epoch boundaries older than the root's slot height? Anything more recent than that will require a leader schedule per fork.

Reply:
@garious I think that is the proposal.  The new schedule is valid for the next epoch, so effectively it is generated for a much much much much older fork then the root.

If the epoch is more then 6 standard deviations longer than the median partition time then the likelihood of a epoch long partition is 1 in 1m.  I think with an epoch of 2 weeks or any human timeframe we are going to get philosophical about partitions.